src/endpoint/routes.ts

```diff
@@ -77,25 +62,25 @@ export default (server, createFactory) => {
     method: 'POST',
     path: '/spawn',
     handler: async (request) => {
-      const opts = request.payload || {}
+      const opts = request.payload ?? {}
       try {
         const ipfsd = await createFactory()
         const id = nanoid()
         // @ts-expect-error opts is a json object
         nodes[id] = await ipfsd.spawn(opts)
         return {
           id: id,
-          apiAddr: nodes[id].apiAddr ? nodes[id].apiAddr.toString() : '',
-          gatewayAddr: nodes[id].gatewayAddr ? nodes[id].gatewayAddr.toString() : '',
-          grpcAddr: nodes[id].grpcAddr ? nodes[id].grpcAddr.toString() : '',
+          apiAddr: nodes[id].apiAddr?.toString(),
+          gatewayAddr: nodes[id].gatewayAddr?.toString(),
+          grpcAddr: nodes[id].grpcAddr?.toString(),
```

Comment:
we're changing the results from being `''` (empty string) when undefined, to being `undefined`. Is this okay?